codalab/client/json_api_client.py

```diff
@@ -14,8 +14,11 @@ def wrapper(*args, **kwargs):
             try:
                 return f(*args, **kwargs)
             except urllib.error.HTTPError as e:
+                # Standardize the data type of error_body generated from different objects (e.g. bottle.HTTPError or
+                # urllib.error.HTTPError) to string. The data type of error message generated by
+                # urllib.error.HTTPError is string by default. The one generated by bottle.HTTPError is bytes.
+                error_body = e.read().decode() if isinstance(e.read(), bytes) else e.read()
```

Comment:
We have two resources that generate this exception. One is from `bottle.HTTPError`. Another is from `urllib.error.HTTPError`. The type of error message generated by `bottle.HTTPError` is bytes. The other one is string. We can either choose to handle the different data types at here or raise the same `HTTPError` throughout our codebase. Open to discuss this.

Reply:
Seems like this should be in another PR?

Reply:
I think you want to call `e.read()` only once.

Reply:
Will create a different PR.